### PR TITLE
Add workflow to introspection branch and bug fix

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Add Github Actions workflow for `introspection` branch, running tests in `"com.mysql.cj.jdbc.*"` and fix flaky multithreaded test.

### Additional Reviewers
@sergiyv-bitquill 
@ColinKYuen 
@brunos-bq

### CODE REVIEW

File: .github/workflows/introspection.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  pull_request:
+    branches:
+      - introspection
+
+jobs:
+  ubuntu-latest-aurora-build-and-run-failover-tests:
+    name: 'Run Tests in `com.mysql.cj.jdbc'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone Repository
```
Comment (brunos-bq) [inline .github/workflows/introspection.yml:16]: does it need to be quoted?

File: .github/workflows/introspection.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - name: 'Set up JDK 8'
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: 'Run Tests'
```
Comment (brunos-bq) [inline .github/workflows/introspection.yml:24]: which tests? can you either write a comment or put a more specific name?

File: .github/workflows/introspection.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        run: |
+          ./gradlew --no-parallel --no-daemon build test --tests "com.mysql.cj.jdbc.*"
+      - name: Archive junit results
+        if: ${{ always() }}
```
Comment (brunos-bq) [inline .github/workflows/introspection.yml:28]: wondering if you need an 'if' if the condition is always?

File: .github/workflows/introspection.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+on:
+  push:
+    branches:
+      - introspection
+  pull_request:
+    branches:
+      - introspection
+
+jobs:
+  ubuntu-latest-aurora-build-and-run-failover-tests:
```
Comment (brunos-bq) [inline .github/workflows/introspection.yml:12]: Do you just run a specific set of tests?

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java [hunk 5]
```diff
@@ -109,7 +109,6 @@ public void stopMonitoring(MonitorConnectionContext context) {
     final String node = this.threadContainer.getNode(context.getNodeKeys());
 
     if (node == null) {
-      log.logWarn(Messages.getString("DefaultMonitorService.InvalidContext"));
```
Comment (brunos-bq) [inline src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java:112]: Any reason to remove a log message?

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/MultithreadedNodeMonitoringFailoverPluginTest.java [hunk 6]
```diff
@@ -110,9 +114,12 @@ void test_1_executeSuccessful() throws ExecutionException, InterruptedException
     when(context.isNodeUnhealthy()).thenReturn(false);
     final int numConnections = 10;
     final int wantedNumberOfInvocations = NUM_PLUGINS_PER_SERVICE * numConnections;
-
     final List<NodeMonitoringFailoverPlugin> plugins = initPlugins(numConnections);
-    runExecuteAsync(numConnections, MONITOR_METHOD_NAME, plugins);
+    List<CompletableFuture<Object>> threads = runExecuteAsync(numConnections, plugins);
+
+    for (final CompletableFuture<Object> thread : threads) {
+      thread.get();
```
Comment (brunos-bq) [inline src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/MultithreadedNodeMonitoringFailoverPluginTest.java:121]: What happens when you call thread.get() ?

File: .github/workflows/introspection.yml (hunk 4 above)
Comment (karenc-bq) [reply .github/workflows/introspection.yml:12]: Yes, just the unit tests most relevant to the introspection changes.

File: src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java (hunk 5 above)
Comment (karenc-bq) [reply src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/DefaultMonitorService.java:112]: If `node` is null it means we are trying to remove a resource that has already been removed, I think we can just silently ignore and return in this case.

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/MultithreadedNodeMonitoringFailoverPluginTest.java (hunk 6 above)
Comment (karenc-bq) [reply src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/MultithreadedNodeMonitoringFailoverPluginTest.java:121]: .get() blocks and waits for the thread to finish executing.

File: .github/workflows/introspection.yml (hunk 4 above)
Comment (brunos-bq) [reply .github/workflows/introspection.yml:12]: Could it be shortened to "run-failover-tests-on-ubuntu-latest" ?
(There's no mention of aurora anywhere else in the yaml)

Comment (karenc-bq) [reply .github/workflows/introspection.yml:12]: This is to stay consistent with `main.yml`'s naming style

Comment (karenc-bq) [reply .github/workflows/introspection.yml:12]: On second thought, I will change the name. The purpose of this workflow is to ensure correctness while developing and debugging for introspection branch. This workflow will be removed once introspection is merged to the main branch.

**Review by ColinKYuen — APPROVED** [review]

**Pull request merged**